modules/apps/captcha/captcha-rest-impl/src/main/java/com/liferay/captcha/rest/internal/util/CaptchaTokenUtil.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.captcha.rest.internal.util;
+
+import com.liferay.portal.kernel.captcha.CaptchaTextException;
+import com.liferay.portal.kernel.encryptor.EncryptorException;
+import com.liferay.portal.kernel.encryptor.EncryptorUtil;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Date;
+
+/**
+ * @author Loc Pham
+ */
+public class CaptchaTokenUtil {
+
+	public static void checkAnswer(
+			Company company, String captchaToken, String answer)
+		throws Exception {
+
+		String captchaJSONString = EncryptorUtil.decrypt(
+			company.getKeyObj(), captchaToken);
+
+		JSONObject captchaJSONObject = JSONFactoryUtil.createJSONObject(
+			captchaJSONString);
+
+		if (!isValidCaptchaToken(captchaJSONObject)) {
+			throw new Exception("Invalid captcha");
+		}
```

Comment:
I think creating a validateCaptcha() method and throwing an exception(Example: IllegalArgumentException) as soon as a validation error is found would be better.